### PR TITLE
Blog Post: OpenSearch automated build system is now live

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
OpenSearch automated build system is now live
 
### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by gaiksaya** [comment]
Hi @natebower @krisfreedain @bbarani Can you guys review this blog?
Thanks!

**Comment by natebower** [comment]
> Hi @natebower @krisfreedain @bbarani Can you guys review this blog? Thanks!

Hi @gaiksaya, has this gone through Patti Juric in Marketing and a review from a technical writer? If not, you'll need to go through them first, and then you can request a review from me via this form: https://form.asana.com/?k=C8mTcYB8-BAqQXZAy6yshQ&d=8442528107068%20form.

**Comment by gaiksaya** [comment]
> 

Hey @natebower , Patti did review it. Waiting on review from a technical writer. @bbarani Can you help in adding the required reviewer?
Thanks!

**Comment by bbarani** [comment]
@setiah Can you please review and provide your inputs?

File: _posts/2022-09-19-public-jenkins.markdown
Comment (carolxob) [inline _posts/2022-09-19-public-jenkins.markdown:11]:
```suggestion
OpenSearch project deployed automated build, test, and release infrastructure that supports the end-to-end lifecycle of public distributions along with the launch of 2.3.0 version. This includes the migration of OpenSearch Gradle check workflows that are executed on every PR merge, to the public infrastructure. Community users can now view logs by accessing the [build infrastructure](https://build.ci.opensearch.org/) to understand the current state of their workflows and monitor runs as needed.
```

It looks like some text is missing at the beginning. Add "The" to the beginning of the first sentence.

Comment (carolxob) [inline _posts/2022-09-19-public-jenkins.markdown:11]:
```suggestion
This includes the migration of OpenSearch Gradle check workflows that are executed on every PR merge, to the public infrastructure. Community users can now view logs by accessing the [build infrastructure](https://build.ci.opensearch.org/) to understand the current state of their workflows and monitor runs as needed.
```
Suggestion: remove the comma after "PR merge".

File: _posts/2022-09-19-public-jenkins.markdown
Comment (carolxob) [inline _posts/2022-09-19-public-jenkins.markdown:13]:
```suggestion
This build system will act as a continuous integration and continuous deployment (CI / CD) system that triggers the generation of distribution builds along with automated execution of integration, backward compatibility and performance tests for every build. The enhanced process will also automatically publish the artifacts to target distribution channels including but not limited to Docker, S3, cloud front, Maven etc. The plugin developers can on-board their plugin artifacts in to OpenSearch distributions by adding the component to the distribution input manifest file. The onboarding plugin will need to provide information such as GitHub repository url, the branch which needs to be built, what platforms does the plugin support and checks to run on the component such as version check, dependency check, etc. See [example](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.3.0/opensearch-1.3.0.yml#L29-L34). The end to end build process will use the information on the file to build, assemble, test and publish the artifacts to corresponding distribution targets.
```
Change first sentence to "This build system acts as a..."

Change second sentence to "The enhanced process also automatically publishes..."

Change "such as GitHub repository url" to "GitHub repository URL"

Change "what platforms does the plugin support" to "which platforms does the plugin support"

Change "The end to end build process will use" to "The end to end build process uses"

File: _posts/2022-09-19-public-jenkins.markdown
Comment (carolxob) [inline _posts/2022-09-19-public-jenkins.markdown:15]:
```suggestion
The build system is deployed using infrastructure as code that uses AWS CDK. The build infrastructure is completely reproducible and consumes [opensearch-ci](https://github.com/opensearch-project/opensearch-ci) repository as its source. The[readme.md](https://github.com/opensearch-project/opensearch-ci/blob/main/README.md) file in [opensearch-ci](https://github.com/opensearch-project/opensearch-ci) repo provides the instructions to reproduce this infrastructure. External contributors can contribute to the build ecosystem by raising the PR on [this](https://github.com/opensearch-project/opensearch-ci) GitHub repository for any future enhancements, bug fixes and new features. 
```

Should "repo" be fully spelled out? Or is it safe to leave as is?

File: _posts/2022-09-19-public-jenkins.markdown
Comment (carolxob) [inline _posts/2022-09-19-public-jenkins.markdown:21]:
```suggestion
The manual processes also added additional overhead for the development teams to create, own and maintain their own automation build process mimicking, our process. The new system automates all the manual steps involved in creating, managing, maintaining and monitoring the DevOps stack for OpenSearch distribution builds to improve reproducibility along with developer and release velocity.
```

**Review by carolxob — CHANGES_REQUESTED** [review]

File: _posts/2022-09-19-public-jenkins.markdown
Comment (pajuric) [reply _posts/2022-09-19-public-jenkins.markdown:11]: Please include an inline link to the 2.3.0 launch blog here: https://opensearch.org/blog/releases/2022/09/opensearch-2-3-is-ready-for-download/

**Comment by gaiksaya** [comment]
Thank you for reviewing @carolxob @pajuric 
I have made the suggested changes. Let me know if there are more. 
Thanks!

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:19]:
```suggestion
Before now, setting up the test and build infrastructure was a tedious process that required a lot of manual intervention and development effort in order to make changes and updates. Also, the infrastructure used for building and releasing the artifacts was not reproducible, and teams were finding it difficult to track the status of builds due to limited visibility into the build process. This created discrepancies between the development team in terms of building, testing, and debugging the artifact builds during every release. 
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:21]:
```suggestion
The manual processes also required development teams to create, own, and maintain their own automation build process, mimicking our process. The new system automates all the manual steps involved in creating, managing, maintaining, and monitoring the DevOps stack for OpenSearch distribution builds in order to improve reproducibility and developer and release velocity.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:23]:
```suggestion
### OpenSearch CI/CD infrastructure design
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:27]:
```suggestion
### Overview of the Jenkins workflow(s)
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:29]:
```suggestion
All the workflows are managed through source-controlled code located in the [GitHub repository](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins):
```

Comment (natebower) [reply _posts/2022-10-24-public-jenkins.markdown:29]: We should name the repository rather than just referring to "the" repository. Which one?

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:31]:
```suggestion
* [Distribution-build-opensearch](https://build.ci.opensearch.org/job/distribution-build-opensearch/): The all-in-one pipeline used to clone, build, assemble, and store OpenSearch and its related plugin artifacts with one click. It currently supports LINUX tarball and Linux RPM artifacts. We are in the process of adding MacOS tarball and Windows Zip.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:34]:
```suggestion
* [Integ-test](https://build.ci.opensearch.org/job/integ-test/): The integration tests for OpenSearch plugins are initiated and run here.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:35]:
```suggestion
* [Integ-test-opensearch-dashboards](https://build.ci.opensearch.org/job/integ-test-opensearch-dashboards/): The integration tests for OpenSearch-Dashboards plugins are initiated and run here.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:36]:
```suggestion
* [Distribution-promote-artifacts](https://build.ci.opensearch.org/job/distribution-promote-artifacts/): The one-click workflow used to pull, sign, and push OpenSearch and OpenSearch Dashboards artifacts from staging to production, including Linux tarball and Linux RPM.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:40]:
```suggestion
* [Docker-promotion](https://build.ci.opensearch.org/job/docker-promotion/): The workflow used to promote Docker images from the staging registry to both production Docker Hub and Amazon Elastic Container Registry (Amazon ECR).
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:41]:
```suggestion
* [Whitesource-scan](https://build.ci.opensearch.org/job/whitesource-scan/): The workflow used to trigger CVE scanning on a daily basis using the whitesource/Mend toolchain.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:42]:
```suggestion
* [Gradle-check](https://build.ci.opensearch.org/job/gradle-check/): The heavyweight Gradle check workflow that serves the PR checks for the OpenSearch repository.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:48]:
```suggestion
Now that the build, test, and release infrastructure is public, the community can access, view, and contribute to the infrastructure as well as workflows that build the OpenSearch and OpenSearch Dashboards distributions. As a user with **_anonymous_** role, you can view all the available jobs, logs, packaged artifacts, and Gradle checks related to OpenSearch and OpenSearch Dashboards workflows.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:51]:
```suggestion
## Wrapping it up
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Can you please address my comments and changes so that we ensure they are applied correctly? Thanks!

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown:29]: The "Github repository" hyperlink is the repository

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown:35]: It's a typo. Will fix it.

File: _posts/2022-10-24-public-jenkins.markdown
Comment (peterzhuamazon) [inline _posts/2022-10-24-public-jenkins.markdown:25]: Change this to the new dates I think?
@gaiksaya

Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown:25]: Will do! Thanks!

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown]: Done

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown]: It retains the intended meaning

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown]: "the" is correct

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown]: "the" is correct

File: _posts/2022-10-24-public-jenkins.markdown
Comment (gaiksaya) [reply _posts/2022-10-24-public-jenkins.markdown]: "the" is correct

**Comment by gaiksaya** [comment]
Hi @natebower ,
Thank you for reviewing. I made the recommended changes and a few more.
Please take a look.
Thanks!

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:39]:
```suggestion
* [Docker-copy](https://build.ci.opensearch.org/job/docker-copy/): The workflow used to copy Docker images between different repositories and the foundation of the [Docker-promotion](https://build.ci.opensearch.org/job/docker-promotion/) workflow.
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:53]:
```suggestion
The primary goal of this project is to improve the reproducibility and transparency of the OpenSearch build infrastructure. The build, release, and test infrastructure used by the OpenSearch Project is easily reproduced by following the steps documented in the [readme.md](https://github.com/opensearch-project/opensearch-ci/blob/main/README.md) file, which uses IaC to reproduce the infrastructure. Contributors can now contribute to this infrastructure by submitting their code in a pull request to the [opensearch-ci](https://github.com/opensearch-project/opensearch-ci) repository. The community can also contribute to the build, release, and test workflows deployed on this infrastructure by submitting their code in a pull request to the [opensearch-build](https://github.com/opensearch-project/opensearch-build) repository. To learn more, check out the [Getting Started page](https://github.com/opensearch-project/opensearch-ci#getting-started). Additionally, if you find any issues or have questions, feel free to either create an [issue](https://github.com/opensearch-project/opensearch-ci/issues) on GitHub or ask a question on the [forum](https://forum.opensearch.org/c/build/12)!
```

File: _posts/2022-10-24-public-jenkins.markdown
Comment (natebower) [inline _posts/2022-10-24-public-jenkins.markdown:48]:
```suggestion
Now that the build, test, and release infrastructure is public, the community can access, view, and contribute to the infrastructure as well as workflows that build the OpenSearch and OpenSearch Dashboards distributions. As a user with the **_anonymous_** role, you can view all the available jobs, logs, packaged artifacts, and Gradle checks related to OpenSearch and OpenSearch Dashboards workflows.
```

**Comment by gaiksaya** [comment]
Hi @natebower,
I have addressed all the recent improvments. Thanks!

**Review by bbarani — APPROVED** [review]

**Pull request merged**
